src/GraphQL.AotCompilationSample.TypeFirst/GraphQL.AotCompilationSample.TypeFirst.csproj

```diff
@@ -0,0 +1,33 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <OutputType>Exe</OutputType>
+    <TargetFramework>net7.0</TargetFramework>
+    <ImplicitUsings>enable</ImplicitUsings>
+    <Nullable>enable</Nullable>
+    <PublishAot>true</PublishAot>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <None Remove="IntrospectionQuery.graphql" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Microsoft.Extensions.DependencyInjection" Version="7.0.0" />
+    <PackageReference Include="Microsoft.Extensions.Options" Version="7.0.0" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <ProjectReference Include="..\GraphQL.MicrosoftDI\GraphQL.MicrosoftDI.csproj" />
+    <ProjectReference Include="..\GraphQL.StarWars.TypeFirst\GraphQL.StarWars.TypeFirst.csproj" />
+    <ProjectReference Include="..\GraphQL.SystemTextJson\GraphQL.SystemTextJson.csproj" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <EmbeddedResource Include="IntrospectionQuery.graphql" />
+  </ItemGroup>
+  
+  <ItemGroup>
+    <TrimmerRootAssembly Include="GraphQL.StarWars.TypeFirst" />
```

Comment:
What about resulting app size? What is it with and without TrimmerRootAssembly ?

Reply:
Doesn't really matter, does it?  But to answer your question, with/without rooted assembly the exe file is:

| Without | With |
|-|-|
| 18,608,128 bytes | 18,644,992 bytes | 

But without rooting the assembly, most of the CLR types (if not all) get notably trimmed and the application does not work.

Note that auxiliary files are generated (pdb and xml files), but these are not needed to run the program.

Reply:
"Static dependencies" in context are specific members (methods, properties, constructors) or type references (types exclusive of their members) within other libraries.

So, although we are rooting the StarWars.TypeFirst assembly (and all its types and their members), we are not rooting its dependencies (at least, not in the same way that rooting an assembly does).

So GraphQL.dll still gets trimmed.  In fact, there are a very few "static dependencies" within StarWars.TypeFirst - only these (excluding .NET Standard references):
- `GraphQL.ImplementsAttribute`'s constructor, which directly references `GraphQLClrOutputTypeReference<T>`
- `GraphQL.NameAttribute`
- `System.ComponentModel.DescriptionAttribute`
- `GraphQL.IgnoreAttribute`
- `GraphQL.Types.Relay.DataObjects.Connection<TNode>` and `Edge<TNode`, and indirectly `Connection<TNode, Edge<TNode>>`, and some properties on those data models
- `GraphQL.FromServicesAttribute`

So nearly all of GraphQL.NET is trimmed away, unless otherwise referenced within Program.cs, even though StarWars.TypeFirst is rooted.

And Program.cs directly references the public properties and methods of `StarWarsQuery` and `StarWarsMutation`, and the constructor of `StarWarsData`, this indirectly references the following:
- `Human` and `Droid`'s constructor and properties, and `Episodes` enum (due to the `StarWarsData` constructor)
- The properties on `StarWarsCharacter` since they are used by the derived `Human` and `Droid` types
- `IStarWarsCharacter` type but no members (due to a public method in `StarWarsQuery`
- `HumanInput`'s members (due to the contents of `AddCharacter`, referenced by a member of `StarWarsMutation`)

So, it ends up that by rooting the StarWars.TypeFirst assembly, there is actually very little difference.  The only thing left that gets trimmed is:
- `StarWarsData.GetFriends` and `GetCharactersAsync`
- `StarWarsCharacter.GetFriends` and `GetFriendsConnection`

Those 4 members are the 36kb difference.  (as far as I can tell)